proto/buf/alpha/registry/v1alpha1/schema.proto

```diff
@@ -0,0 +1,196 @@
+  rpc Reformat(ReformatRequest) returns (ReformatResponse);
+}
+
+// Module identifies a particular module in the registry.
+message Module {
+  // A user or organization. Required.
+  string owner = 1;
+  // A repository or module name. Required.
+  string repository = 2;
+
+  // A reference to a version. Optional. If present, it may refer to either a
+  // tag or a commit.
+  //
+  // If absent, the latest version for the module identified by the above
+  // owner and repository fields is assumed.
+  string reference = 3;
+}
+
+message GetSchemaRequest {
+  // The module from which the schema is retrieved. Required.
+  Module module = 1;
+
+  // A version in the form of a commit ID (may not be a tag or other reference).
+  // If present, the response should only include a schema if the requested
+  // version resolves to something other than this commit. This can be used
+  // to detect newer versions, when the reference in id is empty/absent, or
+  // to detect changes to a tag.
+  string only_if_not_version = 2;
+
+  // One or more element names. The names may be type names (messages or enums),
+  // service names, or method names. All names must be fully-qualified. If any name
+  // is unknown, the request will fail and no schema will be returned.
+  //
+  // The resulting schema contains these elements and all of their dependencies,
+  // which is enough information for the caller to construct a dynamic message for
+  // any requested types (e.g. requested message types or request and response types
+  // for requested service and method names).
+  repeated string element_names = 5;
+
+  // If true, well-known imports will be included the returned set of files. If
+  // false or not present, these files will omitted from the response (e.g. the
+  // caller is assumed to already know the types they define).
+  //
+  // These files are a subset of the files in the "well-known types" module:
+  // https://buf.build/beta/wellknowntypes
+  //
+  // Specifically, the files that are omitted by default are the following:
+  //  * google/protobuf/any.proto
+  //  * google/protobuf/descriptor.proto
+  //  * google/protobuf/duration.proto
+  //  * google/protobuf/empty.proto
+  //  * google/protobuf/field_mask.proto
+  //  * google/protobuf/struct.proto
+  //  * google/protobuf/timestamp.proto
+  //  * google/protobuf/wrapper.proto
+  //
+  // These include all well-known types that have custom JSON formats and
+  // the types in "descriptor.proto", required for Protobuf reflection.
+  // Callers typically use compiled-in versions of these files.
+  bool include_well_known_imports = 6;
+
+  // If true, the returned schema will be expanded to include custom option
+  // definitions for options present on the requested schema elements.
+  bool include_custom_options = 7;
+
+  // If true, the returned schema will include all known extensions for any
+  // extendable messages among the requested schema elements.
+  bool include_extensions = 8;
+}
```

Comment:
The GetImage version of this endpoint that serves trimmed filedescriptors has 
https://github.com/bufbuild/buf/blob/4b1c53e9b38ee78dbacbf3b1dd3810eab7f0de95/proto/buf/alpha/registry/v1alpha1/image.proto#L54-L55

which is required to be off when using `types` (in this pr `element_names`). Should this endpoint have such an option too, or are we going to fixup the partial image generation to have support for synthetic sourcecodeinfo?

Reply:
The returned schema will not have source code info, which is why this is not an option. No, definitely do not want to try to fixup the sourceodeinfo -- that is non-trivial work, and the benefit is unclear. There may be some user agents doing dynamic RPC that want to show comments for elements. But I think that is something we could work in as a potential future feature, and we'll leave it out for now.